### PR TITLE
Defer closing body before checking HTTP status code

### PR DESCRIPTION
**Pull request by nikpivkin**
This is a quote from the [documentation](https://pkg.go.dev/net/http#example-Get) of the `net/http package`:
> When err is nil, resp always contains a non-nil resp.Body. Caller should close resp.Body when done reading from it.

In the current implementation, if the error code is 404, the response body will not close.

### CODE REVIEW

**Review by gmlewis — APPROVED** [review]
Thank you, @nikpivkin !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Review by vandanrohatgi — APPROVED** [review]
Makes sense. LGTM!

**Comment by gmlewis** [comment]
Thank you, @vandanrohatgi !  ❤️ 
Merging.

**Pull request merged**
